Sources/MapboxCoreNavigation/TileStoreConfiguration.swift

```diff
@@ -0,0 +1,82 @@
+
+import Foundation
+import MapboxCommon
+
+/**
+ Describes `TileStore` setup.
+ 
+ This struct encapsulates logic for handling `default` and `custom` paths as well as providing corresponding `TileStore`s.
+ It also covers differences between tile storages for Map and Navigation data. Tupically, you won't need to configure these and rely on defaults, unless you provide pre-downloaded data withing your app in which case you'll need `custom()` path to point to your data.
+ */
+public struct TileStoreConfiguration {
+    /**
+     Describes filesystem location for tile storage folder
+     */
+    public enum Location {
+        /**
+         Encapsulated default location.
+         
+         `rawValue` for this case will return `nil`
+         */
+        case `default`
+        /**
+         User-provided path to tile storage folder
+         */
+        case custom(URL)
```

Comment:
Using `URL` type is problematic because we cannot exactly say that `nil == default` value. Map and Navigator treat `nil` differently: Navigator will switch to a default location while Map may have no tile storage at all.
IMHO in this case an enum with associated value is more descriptive than a struct with static constant.